runner/src/main.rs

```diff
@@ -169,17 +174,38 @@ fn run_examples(opt: &RunExamples) -> Step {
                 None => true,
             })
             .filter(|example| {
-                example.applications.is_empty()
-                    || example
-                        .applications
-                        .get(opt.application_variant.as_str())
-                        .is_some()
+                example
+                    .applications
+                    .empty_or_has_application_variant(opt.application_variant.as_ref())
```

Comment:
(optional) Personally I preferred the previous explicit check for empty or the specific application variant, now this function is just doing two things instead of one.

Reply:
Good point. Done.